crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -783,7 +783,14 @@ impl SlidingSync {
         }
 
         // Force invalidation of all the sticky parameters.
-        let _ = self.inner.sticky.write().unwrap().data_mut();
+        {
+            let mut sticky = self.inner.sticky.write().unwrap();
+            let data = sticky.data_mut();
+
+            // Clear all room subscriptions: we don't want to resend all room subscriptions
+            // when the session will restart.
+            data.room_subscriptions.clear();
+        }
```

Comment:
Can you get rid of all the temporaries, and remove the comment about forcing invalidation? Calling `data_mut()` invalidates sticky parameters anyways.

Reply:
Good to me.